packages/calypso-analytics/src/tracks.ts

```diff
@@ -233,7 +234,9 @@ export function recordTracksEvent( eventName: string, eventProperties?: any ) {
 
 	// Remove properties that have an undefined value
 	// This allows a caller to easily remove properties from the recorded set by setting them to undefined
-	eventProperties = omitBy( eventProperties, ( prop ) => typeof prop === 'undefined' );
+	eventProperties = Object.fromEntries(
+		Object.entries( eventProperties ).filter( ( [ , val ] ) => typeof val !== 'undefined' )
```

Comment:
The code ensures `eventProperties` is not empty at the beginning of the function.